lib/SIL/IR/SILPrinter.cpp

```diff
@@ -2520,10 +2520,11 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
     PrintOptions QualifiedSILTypeOptions =
         PrintOptions::printQualifiedSILType();
     QualifiedSILTypeOptions.CurrentModule = WMI->getModule().getSwiftModule();
-    *this << "$" << WMI->getLookupType() << ", " << WMI->getMember() << " : ";
+    auto lookupType = WMI->getLookupType();
+    *this << "$" << lookupType << ", " << WMI->getMember() << " : ";
     WMI->getMember().getDecl()->getInterfaceType().print(
         PrintState.OS, QualifiedSILTypeOptions);
-    if (!WMI->getTypeDependentOperands().empty()) {
+    if ((getLocalArchetypeOf(lookupType) || lookupType->hasDynamicSelfType()) && !WMI->getTypeDependentOperands().empty()) {
```

Comment:
The commit message talks about opaque archetypes, what's the connection?

Reply:
The SILPrinter prints the first type dependent operand as explicit operand. I don't know why.
So war it was the assumption that if the lookupType _is_ an opened archetype then there is exactly 1 type dependent operand. But this is not true for opaque return types, because those have opened archetypes in the substitution list. This SILPrinter change "restores" the original behavior that the explicit operand is printed if the lookupType _is_ an opened archetype.